pyproject.toml

```diff
@@ -1,5 +1,172 @@
 [build-system]
-requires = ["setuptools", "wheel"]
+requires = ["setuptools>=61.0", "setuptools-scm[git]>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "eodatasets3"
+description = "Packaging, metadata and provenance for OpenDataCube EO3 datasets"
+readme = "README.md"
+requires-python = ">=3.10"
+license = "Apache-2.0"
+authors = [{name = "Open Data Cube"}]
+dynamic = ["version"]
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Operating System :: OS Independent",
+]
+
+# Core dependencies (formerly install_requires)
+dependencies = [
+    "affine",
+    "attrs>=18.1", # 18.1 adds 'factory' syntactic sugar
+    "boltons",
+    "botocore",  # missing from datacube
+    "boto3",
+    "cattrs",
+    "ciso8601",
+    "click",
+    "defusedxml",
+    "fiona",
+    "h5py",
+    "jsonschema>=4.18", # We want a Draft6Validator
+    "numpy>=1.15.4",
+    "pyproj",
+    "rasterio",
+    "ruamel.yaml<0.18", # Pinning as per original, although >=0.18 is the latest
+    "scipy",
+    "shapely",
+    "structlog",
+    "xarray",
+    "datacube>=1.9.0",
+    "python-rapidjson",
+    "pystac>=1.8.4,<1.12",
+]
+
+[project.urls]
+"Bug Reports" = "https://github.com/opendatacube/eo-datasets/issues"
+"Source" = "https://github.com/opendatacube/eo-datasets"
+
+[project.scripts]
+eo3-validate = "eodatasets3.validate:run"
+eo3-prepare = "eodatasets3.scripts.prepare:run"
+eo3-recompress-tar = "eodatasets3.scripts.recompress:main"
+eo3-package-wagl = "eodatasets3.scripts.packagewagl:run"
+eo3-to-stac = "eodatasets3.scripts.tostac:run"
+
+[project.optional-dependencies]
+# Optional valid-data poly handling methods
+algorithms = ["scikit-image"]
+# The (legacy) prepare scripts
+ancillary = ["checksumdir", "netCDF4"]
+# If packaging ard/wagl.
+wagl = ["h5py"] # h5py is already in core, but list it here for clarity if needed.
+# The original setup.py had 'all' as the union of wagl, ancillary, algorithms
+# And 'test' was defined as the same as 'all'.
+# We need to list the dependencies explicitly here in TOML.
+# Combined dependencies: h5py, checksumdir, netCDF4, scikit-image, deepdiff, mock, pep8-naming, pytest, rio_cogeo, sphinx-autodoc-typehints, sphinx_rtd_theme, pytest-xdist, pytest-cov
+all = [
+    "h5py", # From wagl (and core)
+    "checksumdir", "netCDF4", # From ancillary
+    "scikit-image", # From algorithms
+    # Add test dependencies from setup.py's tests_require, as test was all
+    "deepdiff>=8.0",
+    "mock",
+    "pep8-naming",
+    "pytest",
+    "rio_cogeo",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+    "pytest-xdist",
```

Comment:
This belongs to pytest rather than sphinx, can we use alphabetical sorting within these groups?